app/src/main/java/com/example/intandem/fragments/PostsFragment.java

```diff
@@ -207,18 +354,82 @@ public void done(List<Post> posts, ParseException e) {
+                            try {
+                                customPlace.fetchIfNeeded();
+                                destinations.append("place_id:").append(customPlace.getGPlaceId()).append("|");
+                            } catch (ParseException ex) {
+                                Log.e(TAG, ex.toString());
+                                ex.printStackTrace();
+                            }
+                        }
+                        CustomPlace customPlace = allPosts.get(allPosts.size() - 1).getCustomPlace();
+                        try {
+                            customPlace.fetchIfNeeded();
+                            destinations.append("place_id:").append(customPlace.getGPlaceId());
+                        } catch (ParseException ex) {
+                            Log.e(TAG, ex.toString());
+                            ex.printStackTrace();
+                        }
+
+                        GoogleMapsService googleMapsService = retrofit.create(GoogleMapsService.class);
+                        googleMapsService.getDistanceSearchResult(latitude + "," + longitude,
+                                destinations.toString(),
+                                "driving",
+                                "en",
+                                BuildConfig.MAPS_API_KEY).enqueue(new Callback<DistanceSearchResult>() {
+                            @Override
+                            public void onResponse(Call<DistanceSearchResult> call, Response<DistanceSearchResult> response) {
+                                DistanceSearchResult distanceSearchResult = response.body();
+                                Log.i(TAG, "success getting all the distances");
+                                List<DistanceInfo> elements = distanceSearchResult.getRows().get(0).getElements();
+                                for (int i = 0; i < elements.size(); i++) {
+                                    Log.d(TAG, "" + elements.get(i).getDistance().getValue() / 1000.0);
+                                    if ((elements.get(i).getDistance().getValue() / 1000.0) <= maxDistance) {
+                                        Log.d(TAG, "accepted: " + elements.get(i).getDistance().getValue() / 1000.0);
+                                        filteredDistancePosts.add(allPosts.get(i));
+                                    }
+                                }
+                                allPosts.clear();
+                                allPosts.addAll(filteredDistancePosts);
+                                // we created the list of filtered posts now.
+                                adapter.notifyDataSetChanged();
+                            }
+
+                            @Override
+                            public void onFailure(Call<DistanceSearchResult> call, Throwable t) {
+                                Log.e(TAG, "error getting distance");
+                            }
+                        });
```

Comment:
👍  this is great, im glad we were able to merge both logics into one function.

Now for an extra credit we could break up this now very large method into logical chunks.  Like just splitting out functionality into their own helper methods.